src/payment/strategies/humm/humm-payment-strategy.ts

```diff
@@ -0,0 +1,59 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import { CheckoutStore,  InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { ExternalPaymentStrategy } from '../external';
+
+export default class HummPaymentStrategy extends ExternalPaymentStrategy {
+
+    constructor(
+        _store: CheckoutStore,
+        _orderActionCreator: OrderActionCreator,
+        _paymentActionCreator: PaymentActionCreator,
+        _formPoster: FormPoster
+    ) {
+            super(
+            _store,
+            _orderActionCreator,
+            _paymentActionCreator,
+            _formPoster
+        );
+        }
+
+    async initialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId}));
+        } catch (error) {
+            if (!error.body.provider_data || this._isAdditionalActionRequired(error)) {
+                return Promise.reject(error);
+            }
+
+            return new Promise(() => {
+                this._formPoster.postForm(error.body.additional_action_required.data.redirect_url, JSON.parse(error.body.provider_data));
+            });
+        }
+    }
+
+    async finalize(): Promise<InternalCheckoutSelectors> {
```

Comment:
this is not required, it's already implemented in the parent class and the logic here didn't change